docs/dev/profiling.md

```diff
@@ -7,177 +7,371 @@ redirect_from:
-┌─────────────┴─────────────┐┌─────────────┴─────────────┐
-│         SEQ_SCAN          ││           FILTER          │
-│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-│           exams           ││     prefix(name, 'Ma')    │
-│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-│            sid            ││           EC: 1           │
-│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-│           EC: 3           ││             2             │
-│   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   ││          (0.00s)          │
-│             3             ││                           │
-│          (0.00s)          ││                           │
-└───────────────────────────┘└─────────────┬─────────────┘
-                             ┌─────────────┴─────────────┐
-                             │         SEQ_SCAN          │
-                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-                             │          students         │
-                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-                             │            sid            │
-                             │            name           │
-                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-                             │ Filters: name>=Ma AND name│
-                             │  <Mb AND name IS NOT NULL │
-                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-                             │           EC: 1           │
-                             │   ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
-                             │             2             │
-                             │          (0.00s)          │
-                             └───────────────────────────┘
-```
-
-The output of `EXPLAIN ANALYZE` contains the estimated cardinality (`EC`), the actual cardinality, and the execution time for each operator.
-
-It is also possible to save the query plan to a file, e.g., in JSON format:
-
-```sql
--- All queries performed will be profiled, with output in json format.
--- By default the result is still printed to stdout.
 PRAGMA enable_profiling = 'json';
--- Instead of writing to stdout, write the profiling output to a specific file on disk.
--- This has no effect for `EXPLAIN ANALYZE` queries, which will *always* be returned as query results.
-PRAGMA profile_output = '/path/to/file.json';
+PRAGMA profiling_output = '/path/to/file.json';
+
+PRAGMA custom_profiling_settings='{"CPU_TIME": "false", "EXTRA_INFO": "true", "OPERATOR_CARDINALITY": "true", "OPERATOR_TIMING": "true"}';
+
+SELECT name
+FROM students
+JOIN exams USING (sid)
+WHERE name LIKE 'Ma%';
 ```
 
-> This file is overwritten with each query that is issued. If you want to store the profile output for later it should be copied to a different file.
+The contents of the outputted file:
+```json
+{
+  "operator_timing": 0.000372,
+  "operator_cardinality": 0,
+  "extra_info": {},
+  "query_name": "SELECT name\nFROM students\nJOIN exams USING (sid)\nWHERE name LIKE 'Ma%';",
```

Comment:
Potentially move these two JSON examples to a separate file? What do you think, @szarnyasg?

Reply:
I think this is a good solution :)